tests/qos/test_buffer.py

```diff
@@ -0,0 +1,475 @@
+    if not max_headroom_size:
+        pytest.skip('No max headroom found on port {}, skip'.format(port_to_test))
+
+    original_cable_len = duthost.shell('redis-cli -n 4 hget "CABLE_LENGTH|AZURE" {}'.format(port_to_test))['stdout']
+    original_speed = duthost.shell('redis-cli -n 4 hget "PORT|{}" speed'.format(port_to_test))['stdout']
+    original_profile = 'pg_lossless_{}_{}_profile'.format(original_speed, original_cable_len)
+
+    try:
+        # set to super long cable length
+        logging.info('[config a super long cable length]')
+        duthost.shell('config interface cable-length {} 10000m'.format(port_to_test))
+        time.sleep(20)
+        logging.info('verify the profile isn\'t changed')
+        check_profile(duthost, 'BUFFER_PG_TABLE:{}:3-4'.format(port_to_test), original_profile)
+        duthost.shell('config interface cable-length {} {}'.format(port_to_test, original_cable_len))
+
+        # add additional PG
+        logging.info('[config the cable length on the port]')
+        duthost.shell('config interface cable-length {} 300m'.format(port_to_test))
+        time.sleep(20)
+        logging.info('verify the profile has been changed')
+        expected_profile = 'pg_lossless_{}_{}_profile'.format(original_speed, '300m')
+        check_profile(duthost, 'BUFFER_PG_TABLE:{}:3-4'.format(port_to_test), expected_profile)
+        logging.info('add another PG and make sure the system isn\'t broken')
+        duthost.shell('config interface buffer priority-group lossless add {} {}'.format(port_to_test, '5-7'))
+        time.sleep(20)
+        # we can't say whether this will accumulative headroom exceed the limit, but the system should not crash
+        # leverage sanity check to verify that
+        duthost.shell('config interface buffer priority-group lossless remove {} {}'.format(port_to_test, '5-7'))
+        duthost.shell('config interface cable-length {} {}'.format(port_to_test, original_cable_len))
+
+        # static profile
+        logging.info('[config headroom override to PG 3-4]')
+        duthost.shell('config buffer profile add test-headroom --xon 18432 --xoff 50000 -headroom 68432')
+        duthost.shell('config interface buffer priority-group lossless set {} {} {}'.format(port_to_test, '3-4', 'test-headroom'))
+        time.sleep(20)
+        logging.info('verify the profile is applied')
+        check_profile(duthost, 'BUFFER_PG_TABLE:{}:3-4'.format(port_to_test), 'test-headroom')
+        duthost.shell('config interface buffer priority-group lossless add {} {} {}'.format(port_to_test, '5-7', 'test-headroom'))
+        time.sleep(20)
+        # again, we can't say for sure whether the accumulative headroom exceeding.
+        # just make sure the system doesn't crash
+        duthost.shell('config interface buffer priority-group lossless remove {} {}'.format(port_to_test, '5-7'))
+
+        logging.info('[update headroom override to a lager size]')
```

Comment:
'larger'

Reply:
Please have all logs to user starts with capital word